diff --git a/build/fork/NavigationContainer.js b/build/fork/NavigationContainer.js
index 3db5ea4f3106bff3b3bd5799e41914be47ae5052..1e233a2ec7bec10491786b0e5ceca683cc0bafe7 100644
--- a/build/fork/NavigationContainer.js
+++ b/build/fork/NavigationContainer.js
@@ -41,11 +41,32 @@ function NavigationContainerInner({ direction = react_native_1.I18nManager.getCo
     (0, useDocumentTitle_1.useDocumentTitle)(refContainer, documentTitle);
     (0, imperative_api_1.useImperativeApiEmitter)(refContainer);
     const [lastUnhandledLink, setLastUnhandledLink] = react_1.default.useState();
+    const mountedRef = react_1.default.useRef(false);
+    const hasPendingUnhandledLinkRef = react_1.default.useRef(false);
+    const pendingUnhandledLinkRef = react_1.default.useRef();
+    react_1.default.useEffect(() => {
+        mountedRef.current = true;
+        if (hasPendingUnhandledLinkRef.current) {
+            setLastUnhandledLink(pendingUnhandledLinkRef.current);
+            hasPendingUnhandledLinkRef.current = false;
+        }
+        return () => {
+            mountedRef.current = false;
+        };
+    }, []);
+    const handleUnhandledLinking = react_1.default.useCallback((link) => {
+        if (mountedRef.current) {
+            setLastUnhandledLink(link);
+            return;
+        }
+        pendingUnhandledLinkRef.current = link;
+        hasPendingUnhandledLinkRef.current = true;
+    }, []);
     const { getInitialState } = (0, useLinking_1.useLinking)(refContainer, {
         enabled: isLinkingEnabled,
         prefixes: [],
         ...linking,
-    }, setLastUnhandledLink);
+    }, handleUnhandledLinking);
     const linkingContext = react_1.default.useMemo(() => ({ options: linking }), [linking]);
     const unhandledLinkingContext = react_1.default.useMemo(() => ({ lastUnhandledLink, setLastUnhandledLink }), [lastUnhandledLink, setLastUnhandledLink]);
     const onReadyForLinkingHandling = (0, useLatestCallback_1.default)(() => {
